### PR TITLE
Implement kfold function

### LINKED ISSUE #90: Implement k-fold CV on graphs
Graphs are randomly divided into k equal-sized folds. The model is trained k times, each time using a different combination of k-1 folds as the training set. The model is evaluated on the remaining fold as the validation set, calculating the performance metrics of interest (e.g., accuracy). The performance results from each fold are averaged to obtain an overall assessment of the model's performance.

### PR DESCRIPTION
**Pull request by abelsiqueira**
**Description**

Move @lyashevska's implementation of KFold here.
Create model.fit and model.evaluate functions.
Create `kfold_evalute` function.

**Related issues**:

Closes #90 

**Instructions to review the pull request**

### CODE REVIEW

File: bird_cloud_gnn/cross_validation.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        h_feats (int, optional): The number of hidden features of the model
+        n_splits (int, optional): Number of folds. Defaults to 5.
+        learning_rate (float, optional): Learning rate. Defaults to 0.01.
+        num_epochs (int, optional): Training epochs. Defaults to 20.
+        batch_size (int, optional): Batch size used in the data loaders. Defaults to 512.
+    """
+
+    labels = np.array(dataset.labels)
+    # Initialize a stratified k-fold splitter
+    kfold = StratifiedKFold(n_splits=n_splits, shuffle=True, random_state=42)
```
Comment (bart1) [inline bird_cloud_gnn/cross_validation.py:34]: Is it desirable to fix the stae for the random number generator here, I think we don't do that for radarDataSet it would be good to be consistent

Comment (abelsiqueira) [reply bird_cloud_gnn/cross_validation.py:34]: I created an issue to deal with that separately.
